app/javascript/components/top-navbar/user-options.jsx

```diff
@@ -0,0 +1,139 @@
+            {...props}
+            className="dropdown"
+          >
+            {children}
+          </li>
+        )}
+  >
+    <Dropdown.Toggle
+      componentClass={
+              ({ children, ...props }) => (
+                <a
+                  {...props}
+                  aria-haspopup="true"
+                  data-toggle="dropdown"
+                  className="dropdown-toggle nav-item-iconic"
+                >
+                  {children}
+                </a>
+              )
+          }
+    >
+      <Icon type="pf" name="user" />
+      <p id="username_display" data-userid={currentUser.userid} className="navbar__user-name">{`${currentUser.name} | ${applianceName}`}</p>
+    </Dropdown.Toggle>
+    <Dropdown.Menu>
+      <MenuItem disabled>{sprintf(__('Server: %s'), applianceName)}</MenuItem>
+      { miqGroups.length > 1 ? (
+        <li id="droptownMenuGroup" className="dropdown-submenu pull-left">
+          <Dropdown.Toggle
+            useAnchor
+            noCaret
+            componentClass={
+                  ({ children }) => (
+                    <a href="#">
+                      {children}
+                    </a>
+                  )
+              }
+          >
+                Change Group
+          </Dropdown.Toggle>
+          <Dropdown.Menu className="scrollable-menu">
+            {miqGroups.sort().map(group => (
+              (currentGroup || {}).id === group.id ? (
+                <MenuItem id="current-group" key={group.id} disabled title={__('Currently Selected Group')}>
+                  {`${group.description} (${__('Current Group')})`}
+                </MenuItem>
+              ) : (
+                <MenuItem
+                  id={group.description}
+                  key={group.id}
+                  title={__('Change to this Group')}
+                  onClick={(e) => { e.preventDefault(); miqSparkleOn(); miqToggleUserOptions(group.id); }}
+                >
+                  {group.description}
+                </MenuItem>
+              )))
+                }
+          </Dropdown.Menu>
+        </li>
+      ) : (
+        <MenuItem id="single-group-item" disabled>{(currentGroup || {}).description}</MenuItem>
+      )
+          }
+      {userMenu.map(section => (
+        section.visible && (
+        <React.Fragment key={section.id}>
+          <MenuItem divider />
+          { section.items.map(item => (
+            item.visible && (
+            <MenuItem
+              id={`user-menu-${item.name.toLowerCase()}`}
+              key={item.id}
+              href={item.href}
+              onClick={event => !miqCheckForChanges() && event.preventDefault()}
+            >
+              {item.name}
+            </MenuItem>
+            )
+          ))}
+        </React.Fragment>
+        )
+      ))}
+      <MenuItem divider />
+      <MenuItem
+        id="logout-btn"
+        href="/dashboard/logout"
+        onClick={event => !miqCheckForChanges() && event.preventDefault()}
```

Comment:
And this also needs updating - this time for https://github.com/ManageIQ/manageiq-ui-classic/pull/6065

I suggest

```
(e) => miqCheckForChanges() ? (ManageIQ.logoutInProgress = true) : e.preventDefault()
```
```